File: challenges/3-tipos-de-triangulo/javascript/LauanAmorim/tiposTriangulo.js
let ladoA = parseFloat(process.argv[2]);
let ladoB = parseFloat(process.argv[3]);
let ladoC = parseFloat(process.argv[4]);

function tipoTriangulo(x, y, z)
{

  let descOrder = [x, y, z].sort((a , b) => a - b);
  let A = descOrder[2];
  let B = descOrder[1];
  let C = descOrder[0];

  if (A < 0 || B < 0 || C < 0)
  {
    console.log('VALOR NÃO PODE SER MENOR QUE 0');
  }
  else if (A >= (B + C)) 
  {
    console.log('NAO FORMA TRIANGULO');
  } else if (A**2 < B**2 + C**2)
    {
      console.log('TRIANGULO ACUTANGULO');
    } 
    if(A == B && B == C && C == A)
    {
      console.log('TRIANGULO EQUILATERO');
    } if (A**2 == B**2 + C**2) 
    {
      console.log('TRIANGULO RETANGULO');
    } if (A**2 > B**2 + C**2)
    {
      console.log('TRIANGULO OBTUSANGULO');
    } if ((A == B && B != C) || (A == C && C != B) || (C == B && B != A))
    {
      console.log('TRIANGULO ISOSCELES');
    }
}

tipoTriangulo(ladoA, ladoB, ladoC);

File: challenges/3-tipos-de-triangulo/go/joaomoraski/tiposTriangulo.go
package main

import (
	"fmt"
	"log"
	m "math"
	"os" // acessar os argumentos da linha de comando (igual em C, 0 é o nome)
	"sort"
	"strconv" // conversoes
)

func TiposTriangulo(lados []float64) (resultados []string) {
	sort.Slice(lados, func(i, j int) bool {
		return lados[i] > lados[j]
	})
	const LADOA, LADOB, LADOC = 0, 1, 2
	if lados[LADOA] >= (lados[LADOB] + lados[LADOC]) {
		resultados = append(resultados, "NAO FORMA TRIANGULO")
		return
	}
	if m.Pow(lados[LADOA], 2) == (m.Pow(lados[LADOB], 2) + m.Pow(lados[LADOC], 2)) {
		resultados = append(resultados, "TRIANGULO RETANGULO")
	}
	if m.Pow(lados[LADOA], 2) > (m.Pow(lados[LADOB], 2) + m.Pow(lados[LADOC], 2)) {
		resultados = append(resultados, "TRIANGULO OBTUSANGULO")
	}
	if m.Pow(lados[LADOA], 2) < (m.Pow(lados[LADOB], 2) + m.Pow(lados[LADOC], 2)) {
		resultados = append(resultados, "TRIANGULO ACUTANGULO")
	}
	if lados[LADOA] == lados[LADOB] && lados[LADOB] == lados[LADOC] {
		resultados = append(resultados, "TRIANGULO EQUILATERO")
	} else if lados[LADOA] == lados[LADOB] || lados[LADOB] == lados[LADOC] || lados[LADOC] == lados[LADOA] {
		resultados = append(resultados, "TRIANGULO ISOSCELES")
	}
	return

}

func main() {
	// chamada por argumento
	if len(os.Args) != 4 {
		fmt.Println("Número de argumentos informado é maior ou menor do que o necessário")
		return
	}
	// Atoi igual no C
	ladoA, err := strconv.ParseFloat(os.Args[1], 64)
	if err != nil {
		log.Fatalf("Não foi possivel converter '%v' para um inteiro", os.Args[1])
	}
	ladoB, err := strconv.ParseFloat(os.Args[2], 64)
	if err != nil {
		log.Fatalf("Não foi possivel converter '%v' para um inteiro", os.Args[2])
	}
	ladoC, err := strconv.ParseFloat(os.Args[3], 64)
	if err != nil {
		log.Fatalf("Não foi possivel converter '%v' para um inteiro", os.Args[3])
	}
	lados := []float64{ladoA, ladoB, ladoC}
	for _, resultado := range TiposTriangulo(lados) {
		fmt.Println(resultado)
	}

}
